base/reinterpretarray.jl

```diff
@@ -0,0 +1,67 @@
+"""
+Gives a reinterpreted view (of element type T) of the underlying array (of element type S).
+If the size of `T` differs from the size of `S`, the array will be compressed/expanded in
+the first dimension.
+"""
+struct ReinterpretArray{T,S,N,A<:AbstractArray{S, N}} <: AbstractArray{T, N}
+    parent::A
+    Base.reinterpret(::Type{T}, a::A) where {T,S,N,A<:AbstractArray{S, N}} = new{T, S, N, A}(a)
```

Comment:
this code assumes `isbits(T) && isbits(S)`, it should be asserted here on construction (like we did before)